File: core/aggregates/src/main/java/ru/lapotko/coingoal/core/aggregates/position/Goal.java
package ru.lapotko.coingoal.core.aggregates.position;

import ru.lapotko.coingoal.core.aggregates.valueobjects.*;

record Goal(Long id, Weight weight, FiatPercent sellPrice, FiatCoinPercent sellAmount, FiatCoin holdingsRemain,
            Pnl pnl) {
}

File: presentation/ng-coingoal/src/app/position/model/goal.ts
import FiatCoinPercentValue from "./value/fiat-coin-percent-value";
import FiatCoinValue from "./value/fiat-coin-value";
import FiatPercentValue from "./value/fiat-percent-value";
import {PnlValue} from "./value/pnl-value";

export interface Goal {
  id: number,
  sellPrice: FiatPercentValue,
  sellAmount: FiatCoinPercentValue,
  weight: number,
  holdingsRemain: FiatCoinValue,
  pnl: PnlValue
}
